design-system/src/theme-provider.tsx

```diff
@@ -94,6 +94,16 @@ const ThemeProvider = (props: ThemeProviderProps) => {
     }
   }, [props.theme, props.themeOverrides]);
 
+  // The previous hook call is not executed on the server, so we just
+  // apply the theme directly if we are on the server.
+  if (typeof window === 'undefined') {
+    applyTheme({
+      newTheme: props.theme,
+      parentSelector: parentSelectorRef.current,
+      themeOverrides: props.themeOverrides,
+    });
```

Comment:
The generated css files work fine when imported in the server side rendered app 🙌
👆 This fragment didn't work.
The other issue I found was this warning: 
```
Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
    at ThemeProvider
```

> **_NOTE:_** I created a [demo app](https://github.com/kark/vite-ssr-project) to validate this solution. It's based on `npm init vite-plugin-ssr@latest`. There is a lot of boilerplate 

Perhaps I'm missing something or should have checked differently but just wanted to share my findings.

Reply:
Yeah, I guess we would need to decouple the implementation for client/server side rendering.

I think we don't need to spend time on this, having the css file for the new theme should be good enough for the use case of the docs.

Of course having this SSR compatible would be great but if it's not that straightforward maybe it's ok to leave it as-is for now.

Reply:
Thanks for giving it a try.

I'll check again on Monday.

Reply:
My initial attempt was way too naive and it doesn't make sense.

As suggested by Nicola, I'll keep it simple with the new CSS files and we'll explore SSR support in the future.

Reply:
Thanks for looking into it and giving it a try! ❤️